contracts/MetaBioMarketPlace.sol

```diff
@@ -0,0 +1,115 @@
+/// Thanks for inspiration: https://github.com/dabit3/polygon-ethereum-nextjs-marketplace/
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+
+contract MetaBioMarketPlace is ReentrancyGuard {
+    using Counters for Counters.Counter;
+    Counters.Counter private _itemIds;
+    Counters.Counter private _itemsSold;
+
+    address public owner;
+
+    constructor() {
+        owner = msg.sender;
+    }
+
+    struct MarketItem {
+        uint256 itemId;
+        address nftContract;
+        uint256 tokenId;
+        address payable seller;
+        address payable owner;
+        uint256 price;
+        bool sold;
+    }
+
+    mapping(uint256 => MarketItem) private idToMarketItem;
+
+    event MarketItemCreated(
+        uint256 indexed itemId,
+        address indexed nftContract,
+        uint256 indexed tokenId,
+        address seller,
+        address owner,
+        uint256 price,
+        bool sold
+    );
+
+    event MarketItemSold(uint256 indexed itemId, address owner);
+
+    function createMarketItem(
+        address nftContract,
+        uint256 tokenId,
+        uint256 price
+    ) public payable nonReentrant {
+        require(price > 0, "Price must be greater than 0");
+
+        _itemIds.increment();
+        uint256 itemId = _itemIds.current();
+
+        idToMarketItem[itemId] = MarketItem(
+            itemId,
+            nftContract,
+            tokenId,
+            payable(msg.sender),
+            payable(address(0)),
+            price,
+            false
+        );
+
+        IERC721(nftContract).transferFrom(msg.sender, address(this), tokenId);
+
+        emit MarketItemCreated(
+            itemId,
+            nftContract,
+            tokenId,
+            msg.sender,
+            address(0),
+            price,
+            false
+        );
+    }
+
+    function purchaseMarketItem(address nftContract, uint256 itemId)
+        public
+        payable
+        nonReentrant
+    {
+        uint256 price = idToMarketItem[itemId].price;
```

Comment:
is it better to read item from mapping 1 time only to a struct variable
instead of 3 times like this?

Reply:
Thanks for your suggestion. I have just updated in https://github.com/viefam/metabio-smart-contracts/pull/10/commits/ec2f94ece138bd4940164022603f69177df6dcd7

Reply:
<test slack>

Reply:
Test Slack